R/mol_roi.R

```diff
@@ -0,0 +1,253 @@
+#' Identify records located outside a given region of interest.
+#'
+#' This function flags records that are outside a particular spatial region of 
+#' interest (roi). ROIs can be specified in a variety of raster and vector 
+#' formats.  
+#'
+#' @family space
+#' @param data data.frame. Containing geographical coordinates.
+#' @param roi spatial region of interest. Can be provided as an sf, sfc,
+#' SpatialPolygons, SpatialPolygonsDataFrame, rasterLayer, spatRaster,
+#' or as a file path with extension #' ".shp", ".gpkg", or ".tif".
+#' @param lat character string. The column name with latitude in decimal degrees
+#' and WGS84. Default = "decimalLatitude".
+#' @param lon character string. The column with longitude in decimal degrees and
+#' WGS84. Default = "decimalLongitude".
+#' @param byExtOnly logical. If true, occurrences are only flagged if located
+#' outside the roi extent. If false, occurrences are checked for whether they are
+#' located within the roi polygons or raster mask.
+#' @param maskValue numeric or character. Raster value corresponding to "outside"
+#' the region of interest. Ignored if roi is provided as a vector layer.
+#'
+#' @details This test identifies records outside a particular region of interest
+#' defined by a raster or vector layer
+#'
+#' @return A data.frame containing the column ".outside_roi". Compliant
+#' (TRUE) if coordinate is outside user defined roi; otherwise "FALSE".
+#'
+#' @importFrom sf st_bbox st_as_sf st_as_sfc st_within
+#' @importFrom stringr str_detect
+#' @importFrom terra crs rast geom vect
+#'
+#' @export mol_roi
+#'
+#' @examples
+#' x <- data.frame(
+#'   decimalLatitude = c(1, 2, 3, 5),
+#'   decimalLongitude = c(1, 2, 3, 4)
+#' )
+#'
+#' roi <- sf::st_as_sfc(sf::st_bbox(c(xmin = 0, xmax = 4, ymax = 4, ymin = 0), crs = 4326))
+#' 
+#' mol_roi(
+#'   data = x,
+#'   roi  = roi,
+#'   lat = "decimalLatitude",
+#'   lon = "decimalLongitude"
+#' )
+#'
+
+mol_roi <- 
+  function(data,
+           roi,
+           lat = "decimalLatitude",
+           lon = "decimalLongitude",
+           byExtOnly = FALSE,
+           maskValue = NA){
+    
+    ### Run checks
+    if (!is.data.frame(data)) {
+      stop(deparse(substitute(data)), " is not a data.frame", call. = FALSE)
+    }
+    
+    check_col(data, c(lat, lon))
+    
+    if(!any(c("character",
+              "sf",
+              "sfc",
+              "SpatialPolygons",
+              "RasterLayer",
+              "SpatRaster")) %in% class(roi)){
+      stop("The region of interest is not in a valid format.")
+    }
```

Comment:
Parentheses are wrong here replace by 

 if(!any(c("character",
              "sf",
              "sfc",
              "SpatialPolygons",
              "RasterLayer",
              "SpatRaster") %in% class(roi))){
      stop("The region of interest is not in a valid format.")
    }